### DISCUSSION TITLE
How to create a custom endpoint over a collection?

### DISCUSSION THREAD

**Discussion by abdonrd** [directus/directus#13131] [Archive]
With Directus we have the CRUD endpoints by default.

For example, with a `Post` collection I can do this to get one by id:

```
GET /items/posts/:id
```

And now we want to create a custom endpoint like this to run an action over it:

```
GET /items/posts/:id/sync
```

How can we get it?

Or do I need to create it like this?

```
GET /sync/posts/:id
```

---

Related docs https://docs.directus.io/extensions/endpoints/

**Comment by azrikahar** [comment]
I believe that's not possible, so the `/sync/posts/:id` approach should be the way to go 👍 You can always do `/sync/:collection/:id` as well if you'd like to be more collection-agnostic.
